google-cloud-apis/src/com/google/cloud/tools/intellij/cloudapis/JavaApplicationServiceAccountKeyRuntimeConfigurationProvider.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright 2018 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.tools.intellij.cloudapis;
+
+import com.intellij.execution.ProgramRunnerUtil;
+import com.intellij.execution.RunManager;
+import com.intellij.execution.RunnerAndConfigurationSettings;
+import com.intellij.execution.application.ApplicationConfiguration;
+import com.intellij.execution.application.ApplicationConfigurationType;
+import com.intellij.execution.configurations.RunConfiguration;
+import com.intellij.execution.executors.DefaultRunExecutor;
+import com.intellij.execution.runners.ProgramRunner;
+import com.intellij.execution.util.EnvironmentVariable;
+import com.intellij.openapi.project.Project;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import org.jetbrains.annotations.NotNull;
+
+public class JavaApplicationServiceAccountKeyRuntimeConfigurationProvider
```

Comment:
javadoc

Reply:
trying to understand what kind of run configs this will apply to. typical servlet container run configs (jetty, tomcat, etc) will be of this type? do all of these java run configs have injectable environment variables (with user visible UI)?

Reply:
This is plain Java application run configuration only as I understand. You can see the example of a screenshot. I believe all others (including app servers you mentioned) would have other `ConfigurationType` for which we obtain run configurations (`runManager.getConfigurationSettingsList(ApplicationConfigurationType.getInstance())`). We can add more extension points if we thing this makes sense although in containers/K8s world having support for plain `main()` applications should be good enough?

Reply:
what about Maven / Gradle run configs. Will this work with them?

I'm thinking about spring boot apps as an example here since many users will use the spring-boot build plugins to run locally.

Reply:
I will check those, not sure - the `ApplicationConfigurationType` is the one that launches `main()` class, i.e. the most fundamental Java run type. We can definitely add as many additional extension points with more configs as we need in subsequent PRs.